app-client/pom.xml

```diff
@@ -89,27 +82,11 @@
                 <type>pom</type>
                 <scope>import</scope>
             </dependency>
-            <dependency>
-                <groupId>org.wildfly</groupId>
-                <artifactId>wildfly-ejb-client-bom</artifactId>
-                <version>${version.wildfly}</version>
-                <scope>import</scope>
-                <type>pom</type>
-            </dependency>
         </dependencies>
     </dependencyManagement>
 
     <build>
         <plugins>
-            <!-- Enforce Java X -->
-            <plugin>
-                <artifactId>maven-compiler-plugin</artifactId>
-                <version>${version.compiler.plugin}</version>
```

Comment:
thing is that if we don't specify version of compiler plugin it will default to either 2.3.2 which doesn't properly handle annotation processors or 3.2/3.3 which is broken on recompile.
one option would be use jboss-parent which defines all plugin versions.